client/windows/CHeroBackpackWindow.cpp

```diff
@@ -24,22 +24,20 @@ CHeroBackpackWindow::CHeroBackpackWindow(const CGHeroInstance * hero)
 {
 	OBJECT_CONSTRUCTION_CAPTURING(255-DISPOSE);
 
-	stretchedBackground = std::make_shared<CFilledTexture>("DIBOXBCK", Rect(0, 0, 410, 425));
+	arts = std::make_shared<CArtifactsOfHeroBackpack>(Point(windowMargin, windowMargin));
+	arts->setHero(hero);
+	addSet(arts);
+	quitButton = std::make_shared<CButton>(Point(0, 0), "IOKAY32.def", CButton::tooltip(""), [this]() { close(); }, EShortcut::GLOBAL_RETURN);
+	stretchedBackground = std::make_shared<CFilledTexture>("DIBOXBCK", Rect(0, 0, arts->pos.w + 2 * windowMargin, arts->pos.h + quitButton->pos.h + 3 * windowMargin));
 	pos.w = stretchedBackground->pos.w;
 	pos.h = stretchedBackground->pos.h;
 	center();
-
-
-	arts = std::make_shared<CArtifactsOfHeroBackpack>(/*Point(-100, -170)*/Point(10, 10));
-	arts->setHero(hero);
-	addSet(arts);
-
+	std::iter_swap(GH.createdObj.front()->children.begin(), GH.createdObj.front()->children.begin() + 2);
```

Comment:
Why is this needed?

Reply:
First the CArtifactsOfHeroBackpack and CButton are created, then CFilledTexture. CFilledTexture is at the beginning of GH.createdObj.front()->children and will be displayed on top.

Reply:
Why can't we swap creation order and create background first?
If you don't know window size you can update it later, e.g:
```cpp
stretchedBackground = std::make_shared<CFilledTexture>( ... )
(create rest of widgets)
stretchedBackground->pos = actualPos
```

Reply:
Right. this approach works